cmd/main.go

```diff
@@ -60,3 +63,21 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+// setAutoMaxProcs sets the GOMAXPROCS value based on the binary name.
+// It suppresses logs for CLI binaries and logs the setting for services.
+func setAutoMaxProcs(binaryName string) {
+	isCLI := binaryName == "argocd" ||
```

Comment:
just as small improvement suggestion, not really blocker to merge. I think it would be nice to move this function to utils and also reuse it inside 

`switch binaryName {
	case "argocd", "argocd-linux-amd64", "argocd-darwin-amd64", "argocd-windows-amd64.exe":`
	
it should be pretty small change, also it would be super easy to cover with unit tests

Reply:
why do we need to reuse it inside the switch case for cli when we are already setting those values based on binary name?

Reply:
resolved